source/common/stats/thread_local_store.cc

```diff
@@ -22,6 +22,9 @@
 namespace Envoy {
 namespace Stats {
 
+Counter* mutex_contention_instances;
+Gauge* mutex_contention_wait_cycles;
```

Comment:
I think this would crash in unit tests if there is a mutex contention after teardown of one method and before setup of method2.

But that wouldn't be an issue with global atomics.

Reply:
I think this is a good call. See 2f9191d, which rewrites a lot of this logic with global atomics.